valkey: 8.0.3 -> 8.1.2

FliegendeWurst opened the pull request:
https://github.com/valkey-io/valkey/releases/tag/8.1.0
https://github.com/valkey-io/valkey/releases/tag/8.1.1
https://github.com/valkey-io/valkey/releases/tag/8.1.2

Security fix https://github.com/valkey-io/valkey/pull/2146

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

FliegendeWurst commented:
Huh, why does this rebuild so much? The last update had <10 rebuilds :thinking:

LeSuisse commented:
The rebuild count is likely caused by 1c1e56fb1644fc1411bb9236d747d885e7200d32 and one of the python dep. That means this does not need to go into `staging-24.11` and can target `release-24.11`.

LeSuisse commented:
Hum in fact 24.11 should probably use Valkey 8.0.3  instead of being bumped to 8.1.x. I will open a PR.

LeSuisse commented:
I created #401458 since there are some behavior changes in 8.1 which [are currently restricted](https://github.com/NixOS/nixpkgs/issues/390768#issuecomment-2827074325). We can bump to 8.1 after the branch-off.

leona-ya commented:
Removing the security tag for this one as 8.0.3 is in unstable now.

FliegendeWurst committed:
valkey: 8.0.3 -> 8.1.2

https://github.com/valkey-io/valkey/releases/tag/8.1.0
https://github.com/valkey-io/valkey/releases/tag/8.1.1
https://github.com/valkey-io/valkey/releases/tag/8.1.2

leona-ya commented:
For me these tests always fail:
```
[err]: Active Defrag main dictionary: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[err]: Active Defrag eval scripts: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[err]: Active Defrag big hash: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[err]: Active Defrag big list: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[err]: Active Defrag big set: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[err]: Active Defrag big zset: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[err]: Active Defrag stream: standalone in tests/unit/memefficiency.tcl
Expected '1.00' to be more than '1.4' (context: type proc line 17 cmd {assert_morethan $frag $value} proc ::validate_frag_ratio level 2)
[exception]: Executing test client: ERR CONFIG SET failed (possibly related to argument 'activedefrag') - Active defragmentation cannot be enabled: it requires a server compiled with a modified Jemalloc like the one shipped by default with the source distribution.
ERR CONFIG SET failed (possibly related to argument 'activedefrag') - Active defragmentation cannot be enabled: it requires a server compiled with a modified Jemalloc like the one shipped by default with the source distribution
```

FliegendeWurst commented:
That's another test for active defragmentation which requires the custom jemalloc. I'll add it to the list

FliegendeWurst commented:
Did we switch the default back to redis? (Looking at the rebuild numbers: 162 on Linux)

vcunat commented:
Not sure what's going on here.  Maybe some dependencies got cut on `master..staging`.  Either way, it probably isn't mereable directly to `master`.
